app/helpers/application_helper.rb

```diff
@@ -73,24 +73,61 @@ def alert_container_class_for_flash_type type
 
   def filter_badge(label: nil, model: nil, filter_by: nil, color: nil, title: nil)
     if model.present?
-      url = "/#{model}?#{filter_by}=#{CGI.escape(label)}"
+      query_string = build_query_string(toggle_filter(filter_by, label))
+      url = "/#{model}?#{query_string}"
     end
 
+    applied = get_query_params[filter_by].include? label
     case color
-    when 'gray'
-      classes = 'bg-gray-100 text-gray-800'
-      classes += ' bg-gray-200' if request.fullpath == url
     when 'blue'
       classes = 'bg-blue-100 text-blue-800'
-      classes += ' bg-blue-200' if request.fullpath == url
+      classes += ' bg-blue-200' if applied
     else
       classes = 'bg-indigo-100 text-indigo-800'
-      classes += ' bg-indigo-200' if request.fullpath == url
+      classes += ' bg-indigo-200' if applied
     end
 
     render partial: 'partials/filter-badge', locals: {label: label, url: url, classes: classes, title: title}
   end
 
+  def clear_filter_badge(label: nil, model: nil, filter_by: nil, color: nil, title: nil)
+    query_string = build_query_string(get_query_params.filter{|k, _| k!= filter_by})
+    url = "/#{model}?#{query_string}"
```

Comment:
Otherwise, we'll get an extra `?` when no query has been made.

```suggestion
    url = query_string.present? ? "/#{model}?#{query_string}" : "/#{model}"
```

Reply:
Shouldn't we fix this too?

Reply:
yup fix is here:
https://github.com/sradu/covid-volunteers/pull/59/commits/2b6dff5fec22f2e2468bebe565b85ab64587c983#diff-127a850b7d7b20173ad3fba88f8a25e3R127

Reply:
Oh. Got it.